Adopt async eventSender mouse functions in http/tests/

charliewolfe opened the pull request:
#### 433779cba1dc6f1b14ff1b689a309e26eb2f930f
<pre>
Adopt async eventSender mouse functions in http/tests/
<a href="https://rdar.apple.com/132819433">rdar://132819433</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=277352">https://bugs.webkit.org/show_bug.cgi?id=277352</a>

Reviewed by Alex Christensen.

This is needed to fix these tests with site isolation.

* LayoutTests/http/tests/contentdispositionattachmentsandbox/form-submission-disabled.html:
* LayoutTests/http/tests/contentdispositionattachmentsandbox/resources/referer-header-stripped.js:
(async var):
(onload):
* LayoutTests/http/tests/contentextensions/block-cookies-in-ping.html:
* LayoutTests/http/tests/contentextensions/block-ping-resource-type-ping.html:
* LayoutTests/http/tests/contentextensions/block-ping-resource-type-raw.html:
* LayoutTests/http/tests/contentextensions/block-ping.html:
* LayoutTests/http/tests/contentextensions/hide-on-ping-with-ping-that-redirects.html:
* LayoutTests/http/tests/contentextensions/hide-on-ping.html:
* LayoutTests/http/tests/download/default-encoding.html:
* LayoutTests/http/tests/download/inherited-encoding.html:
* LayoutTests/http/tests/inspector/network/ping-type.html:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation.html:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/local/blob/resources/hybrid-blob-util.js:
(this.runTestsWithDrag): Deleted.
(HybridBlobTestUtil): Deleted.
* LayoutTests/http/tests/local/fileapi/resources/setup-file-input-element-for-drag.js:
(async moveMouseToCenterOfElement):
(moveMouseToCenterOfElement): Deleted.
* LayoutTests/http/tests/local/fileapi/upload-zip-bundle-as-blob.html:
* LayoutTests/http/tests/local/formdata/send-form-data-constructed-from-form.html:
* LayoutTests/http/tests/media/autoplay-if-audio-is-playing.html:
* LayoutTests/http/tests/misc/bubble-drag-events.html:
* LayoutTests/http/tests/misc/drag-not-loaded-image.html:
* LayoutTests/http/tests/misc/drag-over-iframe-invalid-source-crash.html:
* LayoutTests/http/tests/navigation/document-location-click-timeout.html:
* LayoutTests/http/tests/navigation/document-location-click.html:
* LayoutTests/http/tests/navigation/document-location-mouseover.html:
* LayoutTests/http/tests/navigation/no-referrer-reset.html:
* LayoutTests/http/tests/navigation/no-referrer-same-window.html:
* LayoutTests/http/tests/navigation/no-referrer-subframe.html:
* LayoutTests/http/tests/navigation/no-referrer-target-blank.html:
* LayoutTests/http/tests/navigation/ping-attribute/resources/utilities.js:
(async clickElement):
(clickElement): Deleted.
* LayoutTests/http/tests/navigation/redirect-preserves-referrer.html:
* LayoutTests/http/tests/navigation/redirect-to-random-url-versus-memory-cache.html:
* LayoutTests/http/tests/security/clipboard/drag-drop-html-cross-origin-iframe-in-same-origin.html:
* LayoutTests/http/tests/security/contentSecurityPolicy/block-all-mixed-content/insecure-ping-in-iframe.https.html:
* LayoutTests/http/tests/security/cookies/third-party-cookie-blocking-user-action.html:
* LayoutTests/http/tests/security/dataTransfer-set-data-file-url.html:
* LayoutTests/http/tests/security/drag-drop-different-origin.html:
* LayoutTests/http/tests/security/drag-drop-local-file.html:
* LayoutTests/http/tests/security/drag-over-remote-content-iframe.html:
* LayoutTests/http/tests/security/history-username-password.html:
* LayoutTests/http/tests/security/listener/xss-window-onclick-addEventListener.html:
* LayoutTests/http/tests/security/listener/xss-window-onclick-shortcut.html:
* LayoutTests/http/tests/security/resources/drag-drop-allowed.html:
* LayoutTests/http/tests/security/resources/referrer-policy-redirect-link-downgrade.html:
* LayoutTests/http/tests/security/resources/referrer-policy-redirect-link.html:
* LayoutTests/http/tests/security/resources/rel-noreferrer.html:
* LayoutTests/http/tests/site-isolation/mouse-events/context-menu-event-twice.html:
* LayoutTests/http/tests/site-isolation/mouse-events/context-menu-main-frame-event.html:
* LayoutTests/http/tests/site-isolation/mouse-events/mainframe-prevent-default.html:
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe.html:
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe.html:
* LayoutTests/http/tests/ssl/applepay-ams-ui/PaymentRequest.https.html:
* LayoutTests/http/tests/ssl/applepay/PaymentRequest.https.html:
* LayoutTests/platform/mac-site-isolation/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/281638@main">https://commits.webkit.org/281638@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/670f02ff0008466c66b3fb3e67fdf491548595be

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60439 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39790 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12997 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64362 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10970 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47464 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11203 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48928 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7635 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62470 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37068 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52361 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29752 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33765 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9582 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9891 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55650 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9872 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66090 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4372 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9704 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56276 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4391 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52333 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56461 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13451 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3633 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35601 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36684 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37775 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36426 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on LayoutTests/http/tests/security/clipboard/clipboard-file-access.html:
```diff
@@ -126,6 +126,7 @@
 {
+    testRunner.waitUntilDone();
     debug("Dragging no files should return an empty file list (arbitrary implementation detail):");
     testDraggingFiles([]);
```
I think you need to await these too.

charliewolfe replied:
This test is actually wk1 only, I'll undo these changes.

achristensen07 commented on LayoutTests/http/tests/download/inherited-encoding.html:
```diff
@@ -4,17 +4,19 @@
 }
 </script>
-<p>Downloaded file name should be "SU���SS.txt" (letters "CCE" are Cyrillic).</p>
+<p>Downloaded file name should be "SU���SS.txt" (letters "CCE" are Cyrillic).</p>
```
This change should not be made.

charliewolfe committed:
Adopt async eventSender mouse functions in http/tests/
rdar://132819433
https://bugs.webkit.org/show_bug.cgi?id=277352

Reviewed by Alex Christensen.

This is needed to fix these tests with site isolation.

* LayoutTests/http/tests/contentdispositionattachmentsandbox/form-submission-disabled.html:
* LayoutTests/http/tests/contentdispositionattachmentsandbox/resources/referer-header-stripped.js:
(async var):
(onload):
* LayoutTests/http/tests/contentextensions/block-cookies-in-ping.html:
* LayoutTests/http/tests/contentextensions/block-ping-resource-type-ping.html:
* LayoutTests/http/tests/contentextensions/block-ping-resource-type-raw.html:
* LayoutTests/http/tests/contentextensions/block-ping.html:
* LayoutTests/http/tests/contentextensions/hide-on-ping-with-ping-that-redirects.html:
* LayoutTests/http/tests/contentextensions/hide-on-ping.html:
* LayoutTests/http/tests/download/default-encoding.html:
* LayoutTests/http/tests/download/inherited-encoding.html:
* LayoutTests/http/tests/inspector/network/ping-type.html:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation.html:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/local/blob/resources/hybrid-blob-util.js:
(this.runTestsWithDrag): Deleted.
(HybridBlobTestUtil): Deleted.
* LayoutTests/http/tests/local/fileapi/resources/setup-file-input-element-for-drag.js:
(async moveMouseToCenterOfElement):
(moveMouseToCenterOfElement): Deleted.
* LayoutTests/http/tests/local/fileapi/upload-zip-bundle-as-blob.html:
* LayoutTests/http/tests/local/formdata/send-form-data-constructed-from-form.html:
* LayoutTests/http/tests/media/autoplay-if-audio-is-playing.html:
* LayoutTests/http/tests/misc/bubble-drag-events.html:
* LayoutTests/http/tests/misc/drag-not-loaded-image.html:
* LayoutTests/http/tests/misc/drag-over-iframe-invalid-source-crash.html:
* LayoutTests/http/tests/navigation/document-location-click-timeout.html:
* LayoutTests/http/tests/navigation/document-location-click.html:
* LayoutTests/http/tests/navigation/document-location-mouseover.html:
* LayoutTests/http/tests/navigation/no-referrer-reset.html:
* LayoutTests/http/tests/navigation/no-referrer-same-window.html:
* LayoutTests/http/tests/navigation/no-referrer-subframe.html:
* LayoutTests/http/tests/navigation/no-referrer-target-blank.html:
* LayoutTests/http/tests/navigation/ping-attribute/resources/utilities.js:
(async clickElement):
(clickElement): Deleted.
* LayoutTests/http/tests/navigation/redirect-preserves-referrer.html:
* LayoutTests/http/tests/navigation/redirect-to-random-url-versus-memory-cache.html:
* LayoutTests/http/tests/security/clipboard/drag-drop-html-cross-origin-iframe-in-same-origin.html:
* LayoutTests/http/tests/security/contentSecurityPolicy/block-all-mixed-content/insecure-ping-in-iframe.https.html:
* LayoutTests/http/tests/security/cookies/third-party-cookie-blocking-user-action.html:
* LayoutTests/http/tests/security/dataTransfer-set-data-file-url.html:
* LayoutTests/http/tests/security/drag-drop-different-origin.html:
* LayoutTests/http/tests/security/drag-drop-local-file.html:
* LayoutTests/http/tests/security/drag-over-remote-content-iframe.html:
* LayoutTests/http/tests/security/history-username-password.html:
* LayoutTests/http/tests/security/listener/xss-window-onclick-addEventListener.html:
* LayoutTests/http/tests/security/listener/xss-window-onclick-shortcut.html:
* LayoutTests/http/tests/security/resources/drag-drop-allowed.html:
* LayoutTests/http/tests/security/resources/referrer-policy-redirect-link-downgrade.html:
* LayoutTests/http/tests/security/resources/referrer-policy-redirect-link.html:
* LayoutTests/http/tests/security/resources/rel-noreferrer.html:
* LayoutTests/http/tests/site-isolation/mouse-events/context-menu-event-twice.html:
* LayoutTests/http/tests/site-isolation/mouse-events/context-menu-main-frame-event.html:
* LayoutTests/http/tests/site-isolation/mouse-events/mainframe-prevent-default.html:
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe.html:
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe.html:
* LayoutTests/http/tests/ssl/applepay-ams-ui/PaymentRequest.https.html:
* LayoutTests/http/tests/ssl/applepay/PaymentRequest.https.html:
* LayoutTests/platform/mac-site-isolation/TestExpectations:

Canonical link: https://commits.webkit.org/281638@main
